javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -35,6 +39,19 @@
         return array;
     }
 
+    @SuppressWarnings("unchecked")
+    static <T> Object[] asArray(Iterable<? extends T> iterable) {
+        if (iterable instanceof Collection<?>) {
+            final Collection<? extends T> collection = (Collection<? extends T>) iterable;
+            return collection.toArray();
+        } else if (iterable instanceof Traversable) {
+            final Traversable<T> traversable = asTraversableAgain(iterable);
```

Comment:
please do a one-liner like in reverseIterator() and delete the asTraversableAgain method completely:

```java
} else if (iterable instanceof Traversable && !(iterable instanceof Iterator)) { // TODO: remove second check with 3.0.0
    final Traversable<T> traversable = (Trabersable<T>) iterable;
```